presto-spi/src/main/java/com/facebook/presto/spi/block/EncoderUtil.java

```diff
@@ -97,4 +100,41 @@ public static void encodeNullsAsBits(SliceOutput sliceOutput, Block block)
 
         return Optional.of(valueIsNull);
     }
+
+    /**
+     * Decode the bit stream created by encodeNullsAsBits.  The valueIsNull buffer must be exactly sized to the position
+     * count.  Returns the number of nulls seen during deserialization.
+     */
+    public static int decodeNullBits(SliceInput sliceInput, boolean[] valueIsNull)
```

Comment:
Do we want to make the `decodeNullBits` method at line 71 to call this method?

Reply:
There's a very slight performance regression in decoding the null bits for array and map blocks if we combine them...

Reply:
@tdcmeehan : hmm.. why ? These two methods are slightly different? Then maybe adding comment ? ;)